### PR TITLE
Feat/tracking refactor

### PR DESCRIPTION
**Pull request by marian-buc**
## What?
As part of the maintenance day tracking refactor I've:
- removed quepasa/TrackAction and added a local useSegmentTracking hook
- added useTrackSignedCustomerEvent, useTrackPageViewEvent hooks
- added error handling for direct apolloClient.query
- cleaned up the folder structure for the tracking setup, marked deprecated functions

## Why?
Streamlined tracking experience

**Ticket(s): [GRW-1055,GRW-1056,GRW-1057,GRW-1058,GRW-1059]**

### CODE REVIEW

File: src/client/containers/SessionContainer.tsx [hunk 1]
```diff
@@ -152,9 +152,8 @@ export const SessionContainer: React.SFC<SessionContainerProps> = ({
             ) {
               await setupSession(
                 {
+                  ...realApolloClient!,
                   client,
-                  subscriptionClient: realApolloClient!.subscriptionClient,
-                  httpLink: realApolloClient!.httpLink,
```
Comment (robinandeer) [inline src/client/containers/SessionContainer.tsx:157]: Why this? Doesn't seem related to tracking, no?

File: src/client/utils/tracking/hooks/useTrackSignedCustomerEvent.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const { isoLocale } = useCurrentLocale()
+
+  const variation = useVariation()
+
+  const { quoteCartId } = useQuoteCartIdFromUrl()
+
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+
+  const trackEventHandler = useCallback(
+    (memberId: TrackSignedEventParams['memberId']) => {
```
Comment (robinandeer) [inline src/client/utils/tracking/hooks/useTrackSignedCustomerEvent.ts:27]: nit: consider passing an object here like the other tracking hooks

**Review by robinandeer — COMMENTED** [review]
Major work indeed! :D

It's just hard to keep track of things when you try to review so give us some time haha

File: src/client/containers/SessionContainer.tsx (hunk 1 above)
Comment (marian-buc) [reply src/client/containers/SessionContainer.tsx:157]: since I added a new field on ApolloClientUtils needed to add it here as well

Comment (marian-buc) [reply src/client/containers/SessionContainer.tsx:157]: also, we're running client.mutate in some places, maybe it would be useful to also wrap that at some point

File: src/client/apolloClient.ts [hunk 3]
```diff
@@ -80,5 +86,20 @@ export const apolloClient = (() => {
...
+    try {
+      const apolloQuery = await client.query<T>(options)
+      if (apolloQuery.error) {
+        datadogRum.addError(apolloQuery.error, options)
+        return
+      } else return apolloQuery.data
+    } catch (e) {
+      datadogRum.addError(e, options)
+      return
+    }
```
Comment (robinandeer) [inline src/client/apolloClient.ts:101]: Do we need to send any more context to datadog here? Or maybe it's gonna get picked up automatically

File: src/client/clientEntry.tsx [hunk 4]
```diff
@@ -28,6 +27,8 @@ Sentry.init({
   release: process.env.SENTRY_RELEASE,
 })
 
+import { apolloClient } from './apolloClient'
```
Comment (robinandeer) [inline src/client/clientEntry.tsx:30]: Hmm why is this moved? To catch errors during start up?

File: src/client/pages/Offer/CheckoutSuccessRedirect.tsx [hunk 5]
```diff
@@ -17,33 +17,26 @@ export const CheckoutSuccessRedirect: React.FC<Props> = ({
...
+        currency: bundle.bundleCost.monthlyNet.currency,
+        total: Number(bundle.bundleCost.monthlyNet.amount),
+        products: [
+          {
+            name: quoteBundleTrackingContractType(bundle),
+          },
+        ],
+        ...getUtmParamsFromCookie(),
+      },
+    })
```
Comment (robinandeer) [inline src/client/pages/Offer/CheckoutSuccessRedirect.tsx:35]: We could probably ask if we can just remove segment... @sonnyandersson1 ?

File: src/client/pages/Offer/PageWrapper.tsx [hunk 6]
```diff
@@ -30,12 +29,14 @@ export const PageWrapper = ({
   const [isCustomerServicePhoneNumberEnabled] = useFeature([
     Features.CUSTOMER_SERVICE_PHONE_NUMBER,
   ])
-
+  const trackOfferEvent = useTrackOfferEvent()
   const handleClickPhoneNumber = (status: 'opened' | 'closed') => {
     if (bundle) {
```
Comment (robinandeer) [inline src/client/pages/Offer/PageWrapper.tsx:34]: Do we need this bundle check anymore?

**Review by robinandeer — COMMENTED** [review]
Great work! Just a bit to big to take everything in and wrap my head around 😅 

Do solve the conflicts and we can merge and test so everything is working as expected.

File: src/client/apolloClient.ts (hunk 3 above)
Comment (marian-buc) [reply src/client/apolloClient.ts:101]: we pass the whole query and variables as context, if we'd need to pass more stuff explicitly we'd need it to supply it as a function argument from outside

**Review by robinandeer — APPROVED** [review]
Okey let's merge this @marian-buc!

Sonny also gave us the go ahead to remove segment so let's do that in the next PR :)
